homeassistant/components/template/lock.py

```diff
@@ -76,22 +81,26 @@ class TemplateLock(TemplateEntity, LockEntity):
 
     def __init__(
         self,
-        hass,
-        config,
-        unique_id,
-    ):
+        hass: HomeAssistant,
+        config: dict[str, Any],
+        unique_id: str | None,
+    ) -> None:
         """Initialize the lock."""
         super().__init__(
             hass, config=config, fallback_name=DEFAULT_NAME, unique_id=unique_id
         )
-        self._state = None
+        self._state: str | bool | LockState | None = None
```

Comment:
Side note: Can we fix the use of `self._state`? It should not be a boolean or a string. Only `LockState` or `None`.

Reply:
https://github.com/home-assistant/core/pull/132093